src/infra/dex/okx/mod.rs

```diff
@@ -178,17 +188,63 @@ impl Okx {
                 amount: quote_result.router_result.to_token_amount,
             },
             allowance: dex::Allowance {
-                spender: eth::ContractAddress(quote_result.tx.to),
+                spender: dex_contract_address,
                 amount: dex::Amount::new(quote_result.router_result.from_token_amount),
             },
             gas: eth::Gas(gas),
         })
     }
 
-    async fn quote(&self, query: &dto::SwapRequest) -> Result<dto::SwapResponse, Error> {
+    /// OKX requires signature of the request to be added as dedicated HTTP
+    /// Header. More information on generating the signature can be found in
+    /// OKX documentation: https://www.okx.com/en-au/web3/build/docs/waas/rest-authentication#signature
+    fn generate_signature(
+        &self,
+        request: &reqwest::Request,
+        timestamp: &str,
+    ) -> Result<String, Error> {
+        let mut data = String::new();
+        data.push_str(timestamp);
+        data.push_str(request.method().as_str());
+        data.push_str(request.url().path());
+        data.push('?');
+        data.push_str(request.url().query().ok_or(Error::SignRequestFailed)?);
+
+        let mut mac = Hmac::<Sha256>::new_from_slice(self.api_secret_key.as_bytes())
+            .map_err(|_| Error::SignRequestFailed)?;
+        mac.update(data.as_bytes());
+        let signature = mac.finalize().into_bytes();
+
+        Ok(BASE64_STANDARD.encode(signature))
+    }
+
+    /// OKX Error codes: [link](https://www.okx.com/en-au/web3/build/docs/waas/dex-error-code)
+    fn handle_api_error(code: i64, message: &str) -> Result<(), Error> {
+        Err(match code {
+            0 => return Ok(()),
+            82000 => Error::NotFound, // Insufficient liquidity
+            82104 => Error::NotFound, // Token not supported
+            50011 => Error::RateLimited,
+            _ => Error::Api {
+                code,
+                reason: message.to_string(),
+            },
+        })
+    }
+
+    async fn send<T, U>(&self, endpoint: &str, query: &T) -> Result<U, Error>
```

Comment:
nit:  The function signature can be improved to clearly state what it really does without requiring the reader to dive deeper into the implementation details.
```suggestion
    async fn send_get_request<T, U>(&self, api_path: &str, query: &T) -> Result<U, Error>
```